torch/csrc/jit/passes/quantization/finalize.cpp

```diff
@@ -22,10 +28,30 @@ graph(%a_dequant, %w_quant, %b):
         %w_dequant = aten::dequantize(%w_quant_unpacked)
         %r = aten::linear(%a_dequant, %w_dequant, %b_unpacked)
         return (%r) )";
+  std::string linear_fp16_with_cast = R"(
+graph(%w, %a_dq, %b, %dtype_fp16, %dtype_fp32, %default_param, %non_blocking):
+        %fp16_tensor = aten::to(%w, %dtype_fp16, %default_param, %default_param, %non_blocking)
+        %fp32_tensor = aten::to(%fp16_tensor, %dtype_fp32, %default_param, %default_param, %non_blocking)
+        %r = aten::linear(%a_dq, %fp32_tensor, %b)
+        return (%r) )";
+  std::string linear_fp16_with_prepack = R"(
+graph(%w, %a_dq, %b, %dtype_fp16, %dtype_fp32, %default_param, %non_blocking):
+        %packed_params = quantized::linear_prepack_fp16(%w, %b)
+        %w_unpacked : Tensor, %b_unpacked : Tensor? = quantized::linear_unpack_fp16(%packed_params)
+        %r = aten::linear(%a_dq, %w_unpacked, %b_unpacked)
+        return (%r) )";
+
+  std::vector<PatternReplaceInfo> patterns_and_replacements = {
+      {linear_with_quant, linear_with_quant_prepack},
+      {linear_fp16_with_cast,
+       linear_fp16_with_prepack,
+       {is_fp16_fp32_cast_op}}};
```

Comment:
non_blocking is still not checked

Reply:
I think that param doesn't have to be a specific value, so I omitted that check

Reply:
did we insert the non_blocking = false code? I think it is better to check this as well, I'm not sure how non_blocking = true can be simulated with the swapped pattern

Reply:
Actually non-blocking and copy are checked with the `false` check. The missing one is `memory_format` which should be None - I'll check for that.